demos/http/http_demo_s3_download_multithreaded/http_demo_s3_download_multithreaded.c

```diff
@@ -919,121 +931,152 @@ int main( int argc,
     }
 
     /**************************** Connect. ******************************/
-
-    /* Establish a TLS connection on top of TCP connection using OpenSSL. */
-    if( returnStatus == EXIT_SUCCESS )
+    do
     {
-        /* Attempt to connect to the HTTP server. If connection fails, retry
-         * after a timeout. The timeout value will be exponentially
-         * increased till the maximum attempts are reached or maximum
-         * timeout value is reached. The function returns EXIT_FAILURE if
-         * the TCP connection cannot be established to broker after
-         * the configured number of attempts. */
-        returnStatus = connectToServerWithBackoffRetries( connectToServer,
-                                                          &networkContext );
-    }
+        /* Reset `returnStatus` in case of a retry. */
+        if( ( returnStatus == EXIT_FAILURE ) && ( demoRunCount > 0 ) )
+        {
+            returnStatus = EXIT_SUCCESS;
+        }
```

Comment:
If the URL is invalid before we get to here, then we will still run the rest of the demo, and I don't think we want that to happen.

Reply:
That's a good catch. Thanks. Will address that.